test_get_index_no_platform_with_offline_cache fails on conda-forge CI

### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

`tests/core/test_subdir_data.py::test_get_index_no_platform_with_offline_cache` fails deterministically on **all conda-forge integration CI jobs** with:

```
assert len(SubdirData.query_all("zlib", channels=online_channels)) > 1
AssertionError: assert 0 > 1
 +    where () = SubdirData.query_all('zlib', channels=('conda-forge',))
```

The test (line 118) removes `CONDA_PLATFORM` and `CONDA_PKGS_DIRS` to restore the default pkgs cache but keeps `CONDA_OFFLINE=yes`. It then asserts that an offline `query_all("zlib")` against `context.channels` (which resolves to `('conda-forge',)`) returns results.

This assumes the default pkgs cache already has conda-forge repodata cached from earlier in the test or from CI setup. On conda-forge CI with Miniforge, this cached repodata is absent, so the offline query returns 0 results.

The `defaults` CI variant passes because Miniconda bootstraps `defaults` channel repodata into the pkgs cache during setup.

**Affected jobs:** Every `conda-forge, integration, 2` matrix cell (linux x64, linux ARM, Python 3.13 and 3.14). Visible across multiple PRs (e.g. #16020, #16035).

**Expected behavior:** The test should warm the channel cache with an online query before testing offline reads, rather than assuming pre-cached repodata.

### Conda Info

N/A — CI-only failure.

### Conda Config

conda-forge CI uses `.github/condarc-conda-forge`:
```
channels:
  - conda-forge
```

### Conda list

N/A — CI-only failure.

### Additional Context

- Failing line: `tests/core/test_subdir_data.py:118`
- The comment on line 113 ("If we re-enable the regular cache location, we do find zlib results") describes the intent — but the cache is never populated for conda-forge in CI.
- The test also has a fragile dependency on `context.channels` resolving to the CI channel config.

--- a/news/16041-fix-offline-cache-test
+++ b/news/16041-fix-offline-cache-test
@@ -1,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Fix `test_get_index_no_platform_with_offline_cache` failing on conda-forge CI by warming the repodata cache before testing offline reads. (#16041)
+
+### Deprecations
+
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* <news item>

--- a/tests/core/test_subdir_data.py
+++ b/tests/core/test_subdir_data.py
@@ -96,6 +96,13 @@ def test_get_index_no_platform_with_offline_cache(
     # only works if CONDA_PLATFORM exists in tests/data/conda_format_repo
     # (test will not pass on newer platforms with default CONDA_PLATFORM =
     # 'osx-arm64' etc.)
+
+    # Undo env vars from the loops above so we can fetch repodata online.
+    monkeypatch.delenv("CONDA_OFFLINE")
+    monkeypatch.delenv("CONDA_REPODATA_TIMEOUT_SECS")
+    monkeypatch.delenv("CONDA_PLATFORM")
+    reset_context()
+
     online_channels = context.channels or ["defaults"]
 
     # Warm the repodata cache while still online so offline reads work

--- a/tests/core/test_subdir_data.py
+++ b/tests/core/test_subdir_data.py
@@ -96,6 +96,13 @@ def test_get_index_no_platform_with_offline_cache(
     # only works if CONDA_PLATFORM exists in tests/data/conda_format_repo
     # (test will not pass on newer platforms with default CONDA_PLATFORM =
     # 'osx-arm64' etc.)
+    online_channels = context.channels or ["defaults"]
+
+    # Warm the repodata cache while still online so offline reads work
+    # regardless of which channel the CI environment is configured with.
+    SubdirData._cache_.clear()
+    assert len(SubdirData.query_all("zlib", channels=online_channels)) > 1
+
     monkeypatch.setenv("CONDA_OFFLINE", "yes")
     monkeypatch.setenv("CONDA_PLATFORM", platform)
     monkeypatch.setenv("CONDA_PKGS_DIRS", str(tmp_path))
@@ -104,7 +111,6 @@ def test_get_index_no_platform_with_offline_cache(
 
     local_channel = Channel(join(CHANNEL_DIR_V1, platform))
     offline_channels = [local_channel]
-    online_channels = context.channels or ["defaults"]
 
     # In offline mode with an empty cache, we expect no results from online channels
     assert len(SubdirData.query_all("zlib", channels=offline_channels)) > 0

